lib/ruboty/capistrano/github.rb

```diff
@@ -0,0 +1,18 @@
+require 'octokit'
+
+module Ruboty::Capistrano
+  class Github
+    class << self
+      def client
+        @client ||= Octokit::Client.new(access_token: Ruboty::Capistrano.config.github_access_token)
+      end
+
+      def branch_exist?(repo_name, branch)
+        client.branch(repo_name, branch)
+        true
+      rescue Octokit::NotFound
+        false
+      end
```

Comment:
僕だったら、↓みたいにかくかな。

```
      def branch_exist?(repo_name, branch)
        !!client.branch(repo_name, branch)
      rescue Octokit::NotFound
        false
      end
```

Octokitのコードも、そんな感じみたいですね。

```ruby
module Octokit
  class Client

    # Methods for the Repositories API
    #
    # @see https://developer.github.com/v3/repos/
    module Repositories

      # Check if a repository exists
      #
      # @see https://developer.github.com/v3/repos/#get
      # @param repo [Integer, String, Hash, Repository] A GitHub repository
      # @return [Sawyer::Resource] if a repository exists, false otherwise
      def repository?(repo, options = {})
        !!repository(repo, options)
      rescue Octokit::InvalidRepository
        false
      rescue Octokit::NotFound
        false
      end
```

好みの部分もあるので、おまかせします。